accounting/invoice/utils.py

```diff
@@ -85,22 +106,27 @@ def send_email_with_invoice(template, subject, extra_email_context=None, client_
     template = get_template(template)
     for client_username in client_usernames:
         client = Account.objects.get(user__username=client_username)
-        providers = [rate.provider for rate in client.client_hourly_rates.all()]
+        providers = [rate.provider for rate in client.client_hourly_rates.all() if rate.active]
         for provider in providers:
-            # Copy over certain details from the latest invoice from the proper month to the new invoice.
-            invoice = models.Invoice.objects.filter(
-                provider=provider,
-                client=client,
-                date__month=past.month if draft_invoice else now.month,
-            ).latest('date')
-            if create_invoice:
+            try:
+                invoice = models.Invoice.objects.filter(
+                    provider=provider,
+                    client=client,
+                    date__month=past.month if draft_invoice else now.month,
+                ).latest('date')
+            except models.Invoice.DoesNotExist:
+                # This may be the provider's first invoice.
+                invoice = None
+
+            if create_invoice or not invoice:
                 invoice = models.Invoice.objects.create(
-                    number=invoice.number,
-                    provider=invoice.provider,
-                    client=invoice.client,
-                    extra_text=invoice.extra_text,
-                    extra_image=invoice.extra_image,
-                    template=invoice.template,
+                    number=choices.InvoiceNumberingScheme.increment_value(
+                        invoice.template.numbering_scheme,
+                        value=invoice.number
+                    ) if invoice else choices.InvoiceNumberingScheme.default_default(),
```

Comment:
@UmanShahzad We can also get at the `numbering_scheme` to use via `provider.invoice_template.numbering_scheme`, so it'd be good to use that here instead of falling all the way back to `default_default`.

Reply:
@itsjeyd Oh yes, good point.